ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,214 @@
+            LOGGER.error(
+                "Exception converting query response to JSON: "
+                f"exc={json_exception} response_status_code={response.status_code} response={response.content}"
+            )
+            raise
+
+    def query(self, query):
+        """get the prometheus query result
+
+        Args:
+            query(str): promthetheus query string
+
+        Returns:
+            dict: query result
+        """
+        return self._get_response(query=f"{self.api_v1}/query?query={query}")
+
+    def get_alert(self, alert_name):
+        """
+        Get alert by alert anem if it's an active alert
+
+        Args:
+             alert(str): alert name
+
+        Examples:
+             result = prometheus.get_alert(alert='WatchDog')
+
+        Returns:
+             dict: dictiionary containing alert name as key and alert data as value
+        """
+        response = self.alerts()
+        alert_list = []
+        for alert in response["data"]["alerts"]:
+            if alert and alert["labels"]["alertname"] == alert_name:
+                alert_list.append(alert)
+        if not alert_list:
+            raise NoActiveAlertsFoundError(
+                f"No active alert found for alert: {alert_name}"
+            )
+        return alert_list
+
+    def get_firing_alerts(self, alert_name):
+        """
+        get all the firing alerts from list of active alerts
+        """
+        alert_list = self.get_alert(alert_name=alert_name)
+        firing_alerts = []
+        for alert in alert_list:
+            if alert["state"] == "firing":
+                firing_alerts.append(alert)
+
+        return firing_alerts
+
+    def wait_for_firing_alert_sampler(self, alert_name, timeout=TIMEOUT_10MIN):
+        """
+        Sample output for an alert and raises timeout exception
+
+        Args:
+             alert(str): alert name
+             timeout(int): wait time, default is 10 mins
+
+        Raise:
+             TimeoutExpiredError: if alert is not fired before wait_timeout
+        """
+        sampler = TimeoutSampler(
+            wait_timeout=timeout,
+            sleep=self.scrape_interval,
+            func=self.get_firing_alerts,
+            alert_name=alert_name,
+        )
+        sample = None
+        try:
+            for sample in sampler:
+                if sample:
+                    LOGGER.info(f"Found alert: {alert_name} in firing state.")
+                    return
+
+        except TimeoutExpiredError:
+            LOGGER.error(f"{alert_name} is not fired")
+            raise
+
+    def get_scrape_interval(self):
+        """
+        get prometheus scrap interval
+
+        Returns:
+             int : scrape time interval
+        """
+
+        response = self._get_response(query=f"{self.api_v1}/targets")
+        result = response["data"]["activeTargets"]
+        for item in result:
+            if item and item["labels"]["job"] == "prometheus-k8s":
```

Comment:
Is it guaranteed that this condition will be satisfiedat some point?
if not it might not set `scrape_interval`

Reply:
yes , if prometheus is running this job must exist

Reply:
You will return 30 if the first item is not `prometheus-k8s`
you need to return 30 at the end.

Reply:
If the first item is not "prometheus-k8s" (maybe it's the second one) you will return 30.

Reply:
setting scrape interval to 30 and reurn if it is None